### DISCUSSION TITLE
微调以后推理生成的数据有一直重复生成，

### DISCUSSION THREAD

**Discussion by TongrongHuang** [zai-org/ChatGLM3#1198] [Q&A] [answered]
<img width="827" alt="Snipaste_2024-05-03_16-01-52" src="https://github.com/THUDM/ChatGLM3/assets/135359473/f8aeb0c3-380b-45d1-ade5-d55873cd3d46">
理论上我只需要前面四行的数据即可，但是正确的数据生成结束以后还会接着不断生成重复的数据，这个问题该如何解决，以前在微调完第一代6B的时候也有这种问题，现在微调的是3代的，还是不知道怎么解决，希望大佬提供意见

**Comment by zRzRzRzRzRzRzR** [comment]
检查你的special token有没有正确，是transformer推理吗

**Reply by TongrongHuang** [reply]
我是在自己新建的ipynb文件导入inference_hf.py中的`load_model_and_tokenizer`函数，然后传入`finetune_demo/output/checkpoint-3000`给model
![image](https://github.com/THUDM/ChatGLM3/assets/135359473/68874d5c-57e4-4916-8ee3-77180dc7161c)

**Reply by TongrongHuang** [reply]
我看了一下微调前后的special_tokens都是`special_tokens={'eos_token': '</s>', 'unk_token': '<unk>', 'pad_token': '<unk>'}`，没有发生变化

**Comment by zRzRzRzRzRzRzR** [comment, marked as answer]
尝试增加 user这个special token作为stop id

**Reply by zRzRzRzRzRzRzR** [reply]
后续的代码会重新将special tokenizer id这边进行重构，对齐huggingface代码
